### ISSUE TITLE
`Dense(units=1)` output shape is not `(batch_size, 1)` as expected

### ISSUE THREAD

**Issue by ghost** [keras-team/tf-keras#578]
Here's a model:

    import random
    import string
    
    import tensorflow as tf
    from keras import Input, Model
    from keras.layers import Dense, Dropout
    from transformers import DistilBertTokenizer, TFDistilBertModel
    
    
    def create_model(input_shape, cache_dir, pretrained_model):
        ids = Input(input_shape, dtype='int32', name='input_ids')
        masks = Input(input_shape, dtype='int32', name='attention_mask')
        x0 = dict(input_ids=ids, attention_mask=masks)
        x = TFDistilBertModel.from_pretrained(pretrained_model, cache_dir=cache_dir)(x0)[0]
        x = Dropout(0.2)(x)
        x = Dense(64)(x)
        x = Dropout(0.2)(x)
        output = Dense(1, 'sigmoid')(x)
        return Model(x0, output)

I need the output to be of shape `(100, 1)` however, it's `(100, 512)` in this example:

    pretrained_model = 'distilbert-base-uncased'
    cache_dir = '.'
    tokenizer = DistilBertTokenizer.from_pretrained(
        pretrained_model, cache_dir=cache_dir
    )
    s = string.ascii_letters
    text = [''.join(random.choice(s) for _ in range(10)) for _ in range(100)]
    tokens = tokenizer.batch_encode_plus(
        text,
        add_special_tokens=True,
        truncation=True,
        return_tensors='np',
        padding='max_length',
    )
    model = create_model((None,), cache_dir, pretrained_model)
    preds = model(dict(tokens))
    print(preds.shape)

shape:

    TensorShape([100, 512])
